### PR TITLE
12.0 mig website logo

### PR DESCRIPTION
**Pull request by eslAmer**

### CODE REVIEW

**Comment by ghost** [comment]
I tested the module (odoo 12 ce), all fine.

File: website_logo/views/website_templates.xml [hunk 1]
```diff
@@ -1,13 +1,10 @@
...
 <odoo>
-
-    <template id="layout_logo_show" inherit_id="website.layout_logo_show">
-        <xpath expr="//a//span" position="attributes">
-            <attribute name="class" add="hide" separator=" "/>
-        </xpath>
-        <xpath expr="//a//span" position="after">
-            <img src="/website_logo.png" alt="Website logo" title="Website logo"/>
+     <template id="add_logo_to_website" inherit_id="website.layout_logo_show" customize_show="True" name="Show Logo">
+        <xpath expr="//header//a[hasclass('navbar-brand')]" position="replace">
```
Comment (pedrobaeza) [inline website_logo/views/website_templates.xml]: Please never do replace and use the other technique as less invasive.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: website_logo/models/res_config.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+from odoo import fields, models
```
Comment (nikul-serpentcs) [inline website_logo/models/res_config.py]: @eslAmer IMO Add after Copyright

File: website_logo/models/website.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+from odoo import fields, models
```
Comment (nikul-serpentcs) [inline website_logo/models/website.py]: same here

File: website_logo/views/website_templates.xml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<odoo>
+    <template id="add_logo_to_website" inherit_id="website.layout_logo_show" customize_show="True" name="Show Logo">
+
+        <xpath expr="//header//a[hasclass('navbar-brand')]" position="after">
+
```
Comment (nikul-serpentcs) [inline website_logo/views/website_templates.xml]: remove blank line

**Review by nikul-serpentcs — COMMENTED** [review]
Improve Code

File: website_logo/models/website.py (hunk 3 above)
Comment (eslAmer) [reply website_logo/models/website.py]: done

File: website_logo/views/website_templates.xml (hunk 1 above)
Comment (eslAmer) [reply website_logo/views/website_templates.xml]: done

File: website_logo/views/website_templates.xml (hunk 4 above)
Comment (eslAmer) [reply website_logo/views/website_templates.xml]: done

File: website_logo/models/res_config.py (hunk 2 above)
Comment (eslAmer) [reply website_logo/models/res_config.py]: done

File: website_logo/__manifest__.py [hunk 5]
```diff
@@ -1,26 +1,24 @@
...
+
 {
     'name': 'Website logo',
     'summary': 'Website company logo',
-    'version': '11.0.1.0.0',
-    'category': 'Website',
-    'author': "Agile Business Group, "
+    'version': '12.0.1.0.0',
+    'category': 'website',
+    'author': "EslAmer, "
```
Comment (pedrobaeza) [inline website_logo/__manifest__.py]: This migration hasn't promoted you to co-author. Please put only on contributors

Comment (eslAmer) [reply website_logo/__manifest__.py]: done

**Comment by ghost** [comment]
Bug.

I try to install the module on my productive website. On tested again on a plain installation, there is no error.

On my website, the company logo is next to the placeholder-icon. And i cant access to the setting.
(Website with 4 Domains as different websites)

Error message by access settings.
TypeError: this.state.fields[fieldName] is undefined

https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:1627
Zurückverfolgung
_renderTagLabel@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:1627:626
_renderNode@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:1618:237
_renderNode@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:2791:38
OdooClass.extend/Class.include/</prototype[name]</<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:3540:368
_.collect@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:13:270
_renderGenericTag@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:1608:301
_renderNode@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:1621:8
_renderNode@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:2791:38
OdooClass.extend/Class.include/</prototype[name]</<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:3540:368
_.collect@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:13:270
_renderGenericTag@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:1608:301
_renderNode@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:1621:8
_renderNode@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:2791:38
OdooClass.extend/Class.include/</prototype[name]</<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:3540:368
_.collect@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:13:270
_renderGenericTag@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:1608:301
_renderNode@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:1621:8
_renderNode@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:2791:38
OdooClass.extend/Class.include/</prototype[name]</<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:3540:368
_.collect@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:13:270
_renderGenericTag@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:1608:301
_renderNode@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:1621:8
_renderNode@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:2791:38
OdooClass.extend/Class.include/</prototype[name]</<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:3540:368
_.collect@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:13:270
_renderGenericTag@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:1608:301
_renderNode@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:1621:8
_renderNode@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:2791:38
OdooClass.extend/Class.include/</prototype[name]</<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:3540:368
_.collect@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:13:270
_renderGenericTag@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:1608:301
_renderNode@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:1621:8
_renderNode@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:2791:38
OdooClass.extend/Class.include/</prototype[name]</<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:3540:368
_.collect@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:13:270
_renderGenericTag@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:1608:301
_renderNode@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:1621:8
_renderNode@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:2791:38
OdooClass.extend/Class.include/</prototype[name]</<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:3540:368
_.collect@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:13:270
_renderGenericTag@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:1608:301
_renderNode@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:1621:8
_renderNode@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:2791:38
OdooClass.extend/Class.include/</prototype[name]</<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:3540:368
_.collect@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:13:270
_renderTagForm@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:1626:16
_renderNode@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:1618:237
_renderNode@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:2791:38
OdooClass.extend/Class.include/</prototype[name]</<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:3540:368
_renderView@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:1633:1254
_render@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:1564:173
_render@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:7:657
OdooClass.extend/</prototype[name]</<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:3537:485
start@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:1268:330
OdooClass.extend/</prototype[name]</<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:3537:485
start@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:1599:8
OdooClass.extend/</prototype[name]</<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:3537:485
start@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:3:642
OdooClass.extend/</prototype[name]</<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:3537:485
_widgetRenderAndInsert/<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:3682:47
then/</</<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:802:678
fire@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:796:281
add@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:797:467
then/</<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:802:649
each@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:625:758
then/<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:802:553
Deferred@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:803:189
then@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:802:518
_widgetRenderAndInsert@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:3681:165
appendTo@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:3672:92
start@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:1275:1505
OdooClass.extend/</prototype[name]</<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:3537:485
start@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:1532:1006
OdooClass.extend/</prototype[name]</<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:3537:485
_widgetRenderAndInsert/<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:3682:47
then/</</<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:802:678
fire@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:796:281
add@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:797:467
then/</<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:802:649
each@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:625:758
then/<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:802:553
Deferred@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:803:189
then@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:802:518
_widgetRenderAndInsert@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:3681:165
appendTo@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:3672:92
_startController@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:486:204
_executeAction/<@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:463:20
then/</</<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:802:678
fire@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:796:281
add@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:797:467
then/</<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:802:649
each@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:625:758
then/<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:802:553
Deferred@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:803:189
then@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:802:518
_executeAction@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:462:39
_executeWindowAction/</<@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:504:135
then/</</<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:802:678
fire@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:796:281
fireWith@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:801:198
Deferred/</deferred[tuple[0]]@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:803:56
fire@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:796:281
fireWith@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:801:198
Deferred/</deferred[tuple[0]]@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:803:56
then/</</<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:802:678
fire@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:796:281
fireWith@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:801:198
Deferred/</deferred[tuple[0]]@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:803:56
_createViewController/<@https://domain.xyz/web/content/1492-3719200/web.assets_backend.js:496:786
then/</</<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:802:678
fire@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:796:281
fireWith@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:801:198
Deferred/</deferred[tuple[0]]@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:803:56
fire@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:796:281
fireWith@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:801:198
then/</</<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:802:874
fire@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:796:281
fireWith@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:801:198
updateFunc/<@https://domain.xyz/web/content/1379-0a731e0/web.assets_common.js:804:482
website_logo

**Comment by ghost** [comment]
No logo setting item available.
screenshot_20181120_233120

**Comment by eslAmer** [comment]
@Axel-G are you sure if module is installed and up to date

**Comment by ghost** [comment]
@eslAmer sure. its from your last commit .... look to my sites, the logo is replaced with the placeholder image

**Comment by eslAmer** [comment]
@Axel-G  user and pwd ?

**Comment by ghost** [comment]
Bug: logo is right  side of (invisible) default company logo.
invisible

**Comment by eslAmer** [comment]
@oca-clabot  hello guys i am wonder why the used email is not my email or even one of my listedemail so why every PR i try to do i  get this comment , i am signed the cla and i am also oca member  ??

**Comment by pedrobaeza** [comment]
It's the email which is being used on your git configuration, and it's not even linked to your GitHub account, so yes, the complaint is correct.

**Comment by eslAmer** [comment]
> It's the email which is being used on your git configuration, and it's not even linked to your GitHub account, so yes, the complaint is correct.

my git config email is my github one and i really doesn't know how this email doing any push throw this account !! should i do anything ?

**Comment by pedrobaeza** [comment]
This commit doesn't have it: https://github.com/OCA/website/pull/534/commits/4fdcd2b7a45037594f757567af655fc700f58f0c

**Comment by eslAmer** [comment]
> This commit doesn't have it: [4fdcd2b](https://github.com/OCA/website/commit/4fdcd2b7a45037594f757567af655fc700f58f0c)

same everything !

**Comment by pedrobaeza** [comment]
No, look at the logo on that commit and the last ones. Any way, rebase, flat the commit history (as you have a lot of commits for migration) and make sure it's correct.

**Comment by eslAmer** [comment]
> No, look at the logo on that commit and the last ones. Any way, rebase, flat the commit history (as you have a lot of commits for migration) and make sure it's correct.

if there's any instructions should i follow or just close this PR and Open New One ??

**Comment by pedrobaeza** [comment]
This is more git knowledge than any other thing. You can forced push in this branch if you want.

**Pull request closed without merging**